phpBB/phpbb/search/backend/search_backend_interface.php

```diff
@@ -0,0 +1,196 @@
+	 * @param string $author_name specifies the author match, when ANONYMOUS is also a search-match
+	 * @param array        &$id_ary passed by reference, to be filled with ids for the page specified by $start and $per_page, should be ordered
+	 * @param int $start indicates the first index of the page
+	 * @param int $per_page number of ids each page is supposed to contain
+	 * @return    boolean|int                        total number of results
+	 */
+	public function author_search(string $type, bool $firstpost_only, array $sort_by_sql, string $sort_key, string $sort_dir, string $sort_days, array $ex_fid_ary, string $post_visibility, int $topic_id, array $author_ary, string $author_name, array &$id_ary, int &$start, int $per_page);
+
+	/**
+	 * Returns if phrase search is supported or not
+	 *
+	 * @return bool
+	 */
+	public function supports_phrase_search(): bool;
+
+	/**
+	 * Updates wordlist and wordmatch tables when a message is posted or changed
+	 * Destroys cached search results, that contained one of the new words in a post so the results won't be outdated
+	 *
+	 * @param string $mode contains the post mode: edit, post, reply, quote ...
+	 * @param int $post_id contains the post id of the post to index
+	 * @param string $message contains the post text of the post
+	 * @param string $subject contains the subject of the post to index
+	 * @param int $poster_id contains the user id of the poster
+	 * @param int $forum_id contains the forum id of parent forum of the post
+	 */
+	public function index(string $mode, int $post_id, string &$message, string &$subject, int $poster_id, int $forum_id);
+
+	/**
+	 * Destroy cached results, that might be outdated after deleting a post
+	 * @param array $post_ids
+	 * @param array $author_ids
+	 * @param array $forum_ids
+	 *
+	 * @return void
+	 */
+	public function index_remove(array $post_ids, array $author_ids, array $forum_ids): void;
+
+	/**
+	 * Destroy old cache entries
+	 *
+	 * @return void
+	 */
+	public function tidy(): void;
+
+	/**
+	 * Create fulltext index
+	 *
+	 * @param int $post_counter
+	 * @return array|null array with current status or null if finished
+	 */
+	public function create_index(int &$post_counter = 0): ?array;
+
+	/**
+	 * Drop fulltext index
+	 *
+	 * @param int $post_counter
+	 * @return array|null array with current status or null if finished
+	 */
+	public function delete_index(int &$post_counter = 0): ?array;
+
+	/**
+	 * Returns true if both FULLTEXT indexes exist
+	 *
+	 * @return bool
+	 */
+	public function index_created(): bool;
+
+	/**
+	 * Returns an associative array containing information about the indexes
+	 *
+	 * @return array|false Language string of error false otherwise
+	 */
+	public function index_stats();
+
+	/**
+	 * Display various options that can be configured for the backend from the acp
+	 *
+	 * @return array array containing template and config variables
+	 */
+	public function acp(): array;
```

Comment:
Maybe make the method a bit more verbose? How about something like `get_acp_options()`?